src/main/java/com/kodilla/ecommercee/mapper/OrderMapper.java

```diff
@@ -0,0 +1,33 @@
+package com.kodilla.ecommercee.mapper;
+
+import com.kodilla.ecommercee.domain.orders.Order;
+import com.kodilla.ecommercee.domain.orders.OrderDto;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class OrderMapper {
+    @Autowired
+    ItemMapper itemMapper;
+    @Autowired
+    private UserMapper userMapper;
+
+    public Order orderDtoToOrder(OrderDto orderDto){
+        return new Order(orderDto.getOrderId(), orderDto.getDate(), userMapper.userDtoToUser(orderDto.getUser()), itemMapper.itemDtoListToItemList(orderDto.getItems()), orderDto.getPaid());
```

Comment:
tu też builder i wstrzykiwanie przez konstruktor

Reply:
Gdy dodaje tutaj wstrzykiwanie przez konstruktor to wywala błędy:
Description:

The dependencies of some of the beans in the application context form a cycle:

   orderController (field private com.kodilla.ecommercee.mapper.OrderMapper com.kodilla.ecommercee.controller.OrderController.orderMapper)
┌─────┐
|  orderMapper defined in file [C:\Users\DominikM\IdeaProjects\JDP-1903-04\out\production\classes\com\kodilla\ecommercee\mapper\OrderMapper.class]
↑     ↓
|  itemMapper defined in file [C:\Users\DominikM\IdeaProjects\JDP-1903-04\out\production\classes\com\kodilla\ecommercee\mapper\ItemMapper.class]
└─────┘

Wszędzie indziej działa.

Reply:
A widzisz, to jest circular dependency problem. Bardzo zła rzecz. Jeśli się objawia to znaczy, że co poszło mocno nie tak z zaprojektowaniem systemu. W takim przypadku zazwyczaj okazuje się, że pewne rzeczy są niepotrzebne (np relacje w obie strony, a powinna być w jedną) lub trzeba wynieść wspólne części do nowego serwisu. Poczytaj o tym problemie, bo to bardzo ważna rzecz.

Reply:
Mappery usunięte, brak błędu